File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    message: str
    notification_type: NotificationType
    related_invoice_id: Optional[UUID] = None
    is_read: bool
    created_at: datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    unread_count: int
    total: int

File: frontend/src/types/notification.ts
export type NotificationType = 'DUPLICATE_ALERT' | 'ANOMALY_ALERT' | 'INVOICE_PROCESSED' | 'INVOICE_ERROR' | 'SYSTEM'

export interface NotificationResponse {
  id: string
  user_id: string
  title: string
  message: string
  notification_type: NotificationType
  related_invoice_id: string | null
  is_read: boolean
  created_at: string
}

export interface NotificationListResponse {
  items: NotificationResponse[]
  total: number
  unread_count: number
}
